htmap/maps.py

```diff
@@ -583,6 +583,34 @@ def component_statuses(self) -> List[state.ComponentStatus]:
     def components_by_status(self) -> Mapping[state.ComponentStatus, Tuple[int, ...]]:
         """
         Return the component indices grouped by their states.
+
+        Examples
+        --------
+        This example finds the completed jobs for a submitted map,
+        and processes those results:
+
+        .. code:: python
+
+           from time import sleep
+           import htmap
+
+           def job(x):
+               sleep(x)
+               return 1 / x
+
+           futures = htmap.map(job, [0, 2, 4, 6, 8], tag="foo")
+
+           # Wait for all jobs to finish.
+           # Alternatively, use `futures = htmap.load("foo")` on a different process
+           sleep(10)
+
+           jobs = futures.components_by_status()
+           jobs = {k.name: v for k, v in jobs.items()}  # nicer key names
+           for future in jobs["COMPLETED"]:
+               result = futures.get(future)
+               # Whatever processing needs to be done
+               print(result)  # prints "2", "4", "6", and "8"
```

Comment:
```suggestion
           m = htmap.map(job, [0, 2, 4, 6, 8], tag="foo")

           # Wait for all jobs to finish.
           # Alternatively, use `futures = htmap.load("foo")` on a different process
           sleep(10)

           completed = m.components_by_status()[htmap.JobStatus.COMPLETED]
           for component in completed:
               result = m.get(component)
               # Do whatever processing needs to be done
```

Reply:
I found myself getting annoyed with having to type on `htmap.JobStatus.COMPLETED` when doing the work that led to this example. That's a little nicer. I have a slight preference to keep it `"COMPLETED"` as a key.

Reply:
IIRC, `htmap.JobStatus.COMPLETED` doesn't much information. Is that accurate? Could the keys be changed to `"COMPLETED"` instead?

Reply:
Silly me, I just realized that my fancy `StrEnum` is not actually a `StrEnum`...

Just pushed https://github.com/htcondor/htmap/pull/196/commits/48d6e80dc59f99bcc06e8ddf855b265f74f10b83, you should now be able to use `"COMPLETED"` as the key without even doing the intermediate transformation (and let's do it that way in the example).

(And, of course, I typo'd and the enum is actually called `htmap.ComponentStatus`...)

Reply:
Thanks! That's a lot nicer. I've modified the example. I've left in the `futures` and `jobs` variable names because I find that more clear and consistent with other Python libraries (threading, Dask, multiprocessing, etc).

Reply:
I think I have to insist on the variable names for consistency with the rest of the documentation. I'm open to a discussion about changing the naming scheme, but that should go on a separate issue.